packages/app/src/systems/Network/machines/networksMachine.ts

```diff
@@ -211,6 +211,7 @@ export const networksMachine = createMachine(
             if (!input?.data) {
               throw new Error('Invalid network input');
             }
+            await NetworkService.validateAddNetwork(input);
```

Comment:
I think it'd make sense for validateAddNetwork to also select/return the network just like `addNetwork` does, so that it becomes less redundant and avoids having to use both side by side, specially because `addNetwork` is already called inside `validateAddNetwork`

Reply:
I don't think worth for this scope changing the `validateAddNetwork` behavior. 
We need to have them working separately because we are not running the validation everywhere.

Only 3 places call this method.

2 accessible by the user. 1 called by us.